src/main/java/betterquesting/handlers/SaveLoadHandler.java

```diff
@@ -143,143 +217,139 @@ public void loadDatabases(MinecraftServer server) {
-        NameCache.INSTANCE.readFromNBT(nbt4.getTagList("nameCache", 10), false);
-
-        // === LIVES ===
+        getPlayerProgressFiles().forEach(file -> {
+            JsonObject json = JsonHelper.ReadFromFile(file);
+            NBTTagCompound nbt = NBTConverter.JSONtoNBT_Object(json, new NBTTagCompound(), true);
+            QuestDatabase.INSTANCE.readProgressFromNBT(nbt.getTagList("questProgress", 10), true);
+        });
+    }
 
-        JsonObject j5 = JsonHelper.ReadFromFile(fileLives);
+    private void loadParties() {
+        JsonObject json = JsonHelper.ReadFromFile(fileParties);
 
-        NBTTagCompound nbt5 = NBTConverter.JSONtoNBT_Object(j5, new NBTTagCompound(), true);
-        LifeDatabase.INSTANCE.readFromNBT(nbt5.getCompoundTag("lifeDatabase"), false);
+        NBTTagCompound nbt = NBTConverter.JSONtoNBT_Object(json, new NBTTagCompound(), true);
+        PartyManager.INSTANCE.readFromNBT(nbt.getTagList("parties", 10), false);
+    }
 
-        BetterQuesting.logger.info("Loaded " + QuestDatabase.INSTANCE.size() + " quests");
-        BetterQuesting.logger.info("Loaded " + QuestLineDatabase.INSTANCE.size() + " quest lines");
-        BetterQuesting.logger.info("Loaded " + PartyManager.INSTANCE.size() + " parties");
-        BetterQuesting.logger.info("Loaded " + NameCache.INSTANCE.size() + " names");
+    private void loadNames() {
+        NameCache.INSTANCE.reset();
+        JsonObject json = JsonHelper.ReadFromFile(fileNames);
 
-        MinecraftForge.EVENT_BUS.post(new DatabaseEvent.Load(DBType.ALL));
+        NBTTagCompound nbt = NBTConverter.JSONtoNBT_Object(json, new NBTTagCompound(), true);
+        NameCache.INSTANCE.readFromNBT(nbt.getTagList("nameCache", 10), false);
     }
 
-    public void saveDatabases() {
-        List<Future<Void>> allFutures = new ArrayList<>();
-        // === CONFIG ===
+    private void loadLives() {
+        LifeDatabase.INSTANCE.reset();
+        JsonObject json = JsonHelper.ReadFromFile(fileLives);
 
-        if (isDirty || QuestSettings.INSTANCE.getProperty(NativeProps.EDIT_MODE)) {
-            NBTTagCompound jsonCon = new NBTTagCompound();
+        NBTTagCompound nbt = NBTConverter.JSONtoNBT_Object(json, new NBTTagCompound(), true);
+        LifeDatabase.INSTANCE.readFromNBT(nbt.getCompoundTag("lifeDatabase"), false);
+    }
 
-            jsonCon.setTag("questSettings", QuestSettings.INSTANCE.writeToNBT(new NBTTagCompound()));
-            jsonCon.setTag("questDatabase", QuestDatabase.INSTANCE.writeToNBT(new NBTTagList(), null));
-            jsonCon.setTag("questLines", QuestLineDatabase.INSTANCE.writeToNBT(new NBTTagList(), null));
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private void checkLegacyFiles(File rootDir) {
+        if (new File(rootDir, "QuestDatabase.json").exists() && !fileDatabase.exists()) {
+            File legFileDat = new File(rootDir, "QuestDatabase.json");
```

Comment:
Nomenclature here is awkward. "File" is redundant since that's the type of the variable. Three letter abbreviations are hard to follow. Why not just "legacyQuests", "legacyProgress", etc.?